### ISSUE TITLE
Expose internal state/interesting changes to a parent via a hook

### ISSUE THREAD

**Issue by docccdev** [reactjs/react-autocomplete#129]
Can pass isOpen through the props?

**Comment by CMTegner** [comment]
I don't entirely understand what you're asking. Do you wish to control `state.isOpen` via props? What would be the use-case for this?

**Comment by docccdev** [comment]
Yes exactly. I need it for the parent component to know about the state of isOpen

**Comment by docccdev** [comment]
Parent component changes its appearance on the state.isOpen of autocomplete.

**Comment by CMTegner** [comment]
So something like `props.onMenuVisibilityChange`?

```js
render () {
  return (
    <Autocomplete
      onMenuVisibilityChange={isOpen => this.setState({ isAutocompleteOpen: isOpen })}
    />
  )
}
```

**Comment by docccdev** [comment]
Excellent solution is right for me :)

**Comment by sprjr** [comment]
Given the discussions around #138, it might be better to generalize this, e.g. `onUpdate` or something like that, which would then invoke a single callback and pass a map of the current state values that a composer may wish to know?

**Comment by CMTegner** [comment]
Perhaps. I wonder if there's a common way to expose internal state. Do you know of any other reusable components that do something similar?

**Comment by westongame** [comment]
Hi. Can you tell me pls, in witch version are you planning to implement this feature?

**Comment by CMTegner** [comment]
Closing this since the original request is covered by #163 (published as v1.3).
